File: apps/mobile/src/screens/QuestHubScreen.js
import { Text, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { ChibiButton } from '../components/Chibi';

export default function QuestHubScreen({ navigation }) {
  return (
    <SafeAreaView className="flex-1 bg-background px-margin-mobile pt-md gap-4">
      <Text className="text-2xl font-bold text-on-background mb-2">Quests</Text>

      <ChibiButton className="p-4 items-start" onPress={() => navigation.navigate('PreGameReady', { mode: 'run' })}>
        <Text className="text-lg font-bold text-on-primary-container">🗡️ Today's Quest</Text>
        <Text className="text-on-primary-container">Track a run with live GPS, distance, and pace</Text>
      </ChibiButton>

      <ChibiButton className="p-4 items-start" onPress={() => navigation.navigate('PreGameReady', { mode: 'quest_game' })}>
        <Text className="text-lg font-bold text-on-primary-container">🏆 Rank Match</Text>
        <Text className="text-on-primary-container">Arm-swing arcade — compete for XP on the leaderboard</Text>
      </ChibiButton>

      <View className="flex-1" />
    </SafeAreaView>
  );
}

File: fitQuest/fitQuest/Screens/QuestHubScreen.swift
//
//  QuestHubScreen.swift
//  fitQuest
//
//  Ported from screens/QuestHubScreen.js.
//

import SwiftUI

struct QuestHubScreen: View {
    @EnvironmentObject private var router: Router

    var body: some View {
        VStack(spacing: Tokens.Spacing.base) {
            Text("Quests")
                .font(.title.bold())
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(.bottom, Tokens.Spacing.sm)

            Button {
                router.push(.preGameReady(mode: .run))
            } label: {
                VStack(alignment: .leading, spacing: 4) {
                    Text("🗡️ Today's Quest").font(.title3.bold())
                    Text("Track a run with live GPS, distance, and pace")
                        .font(.subheadline)
                }
                .foregroundStyle(Tokens.Colors.onPrimaryContainer)
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(Tokens.Spacing.sm)
            }
            .buttonStyle(.chibi)

            Button {
                router.push(.preGameReady(mode: .questGame))
            } label: {
                VStack(alignment: .leading, spacing: 4) {
                    Text("🏆 Rank Match").font(.title3.bold())
                    Text("Arm-swing arcade — compete for XP on the leaderboard")
                        .font(.subheadline)
                }
                .foregroundStyle(Tokens.Colors.onPrimaryContainer)
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(Tokens.Spacing.sm)
            }
            .buttonStyle(.chibi)

            Spacer()
        }
        .padding(Tokens.Spacing.md)
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .top)
        .background(Tokens.Colors.background)
    }
}

#Preview {
    QuestHubScreen()
        .environmentObject(Router())
}
